### PR TITLE
Fix generated_plugin_registrant.dart not being ignored

### PR DESCRIPTION
**Pull request by bartekpacia**
Wow, finally! I've lost a few hours in the last few days tracking a very strange error with this package, which led to my GitHub Action (which was using `import_sorter` to check if imports are sorted) failing for seemingly no reason.

### What was wrong
`generated_plugin_registrant.dart` file was not being ignored by import_sorter - the logic was flawed. We want to check if `dependencies` in `pubspec.yaml` contain `flutter`, not `flutter/`!

Also, if `import_sorter` failed, it now prints which file caused that.

Now it's fixed and works just fine. Please release `v4.5.2` ASAP, I want to be able to use it finally in my GH Actions :D

### Trivia
**Why were my GitHub Actions failing?**
Because I've set up `import_sorter` to run automatically on save on my computers. It formatted all files, including `generated_plugin_registrant.dart`. But on GitHub Actions, after `generated_plugin_registrant.dart` was generated, nothing was sorting its imports, so this tool was failing.

I hope I described it clearly and that nobody before had lost theirs time because of this bug:)

### CODE REVIEW

File: bin/main.dart [hunk 1]
```diff
@@ -67,11 +67,17 @@ void main(List<String> args) {
 
   // Getting all the dart files for the project
   final dartFiles = files.dartFiles(currentPath, args);
-  if (dependencies.contains('flutter/') &&
-      dartFiles
-          .containsKey('${currentPath}/lib/generated_plugin_registrant.dart')) {
+  final containsFlutter = dependencies.contains('flutter');
```
Comment (bartekpacia) [inline bin/main.dart:70]: This PR could basically be just changing `flutter/` to `flutter`, but I decided to improve some other parts a bit as well.

File: bin/main.dart [hunk 2]
```diff
@@ -67,11 +67,17 @@ void main(List<String> args) {
...
+  final containsRegistrant = dartFiles
+      .containsKey('${currentPath}/lib/generated_plugin_registrant.dart');
+
+  stdout.writeln('contains flutter: ${containsFlutter}');
+  stdout.writeln('contains registrant: ${containsRegistrant}');
```
Comment (bartekpacia) [inline bin/main.dart:75]: Feel free to remove this after merge, I added this so it's more clear.

**Comment by bartekpacia** [comment]
Another nice solution would be removing all logic related to removing zgenerated_plugin_registrant.dart` and just advising users to add it to `ignored_files` in `pubspec.yaml`.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+  "dart.lineLength": 80
+}
```
Comment (Matt-Gleich) [inline .vscode/settings.json:3]: Please remove this file

Comment (Matt-Gleich) [reply .vscode/settings.json:3]: Actually see a comment I made later please

File: bin/main.dart [hunk 4]
```diff
@@ -90,7 +96,13 @@ void main(List<String> args) {
...
 
     final sortedFile = sort.sortImports(
-        file.readAsLinesSync(), packageName, emojis, exitOnChange, noComments);
+      file.readAsLinesSync(),
+      packageName,
+      emojis,
+      exitOnChange,
+      noComments,
+      filePath: filePath,
+    );
```
Comment (Matt-Gleich) [inline bin/main.dart:105]: Is this from that vscode settings file? If so feel free to keep the file.

File: lib/sort.dart [hunk 5]
```diff
@@ -156,8 +154,10 @@ ImportSortData sortImports(
   final sortedFile = sortedLines.join('\n');
   final original = lines.join('\n') + '\n';
   if (exitIfChanged && original != sortedFile) {
-    stdout.write('\n┗━━🚨 ');
-    stdout.write('Please run import sorter!'.bold().red());
+    if (filePath != null) {
+      stdout.writeln(
+          '\n┗━━🚨 File ${filePath} does not have its imports sorted.');
```
Comment (Matt-Gleich) [inline lib/sort.dart:159]: Please bring back the red and bold display. It's most likely a problem with your terminal's display colors/system if the color isn't being displayed correctly. It works fine for me.

**Review by Matt-Gleich — CHANGES_REQUESTED** [review]
Hey @bartekpacia! Thanks for the work on this. Just a few little changes and then we should be good to merge.

File: bin/main.dart (hunk 4 above)
Comment (bartekpacia) [reply bin/main.dart:105]: No, it's me – I almost always (when it makes sense) add the trailing comma because it makes the code prettier :)

Comment (bartekpacia) [reply bin/main.dart:105]: I'll remove the whole `.vscode` directory and add it to `.gitignore`, but then it'd be nice to include some info in `README.md` or `CONTRIBUTING.md` that this project uses lineLength = 80

File: lib/sort.dart (hunk 5 above)
Comment (bartekpacia) [reply lib/sort.dart:159]: I didn't customize anything - just a default Ubuntu terminal (test it out for yourself – it's available on GitHub Actions). I think that many people are using Ubuntu and may encounter this.

Comment (Matt-Gleich) [reply lib/sort.dart:159]: Ok, we can cut it then

File: bin/main.dart (hunk 4 above)
Comment (Matt-Gleich) [reply bin/main.dart:105]: That would be great, thank you!

**Review by Matt-Gleich — APPROVED** [review]
Looks good to me! Thanks for this!

**Comment by Matt-Gleich** [comment]
This is now out in import_sorter v4.6.0. Thanks for the PR!

**Pull request merged**